extensions/vertx-http/runtime/src/main/java/io/quarkus/devui/runtime/comms/JsonRpcRouter.java

```diff
@@ -233,8 +234,22 @@ private void route(JsonRpcRequest jsonRpcRequest, ServerWebSocket s) {
             }
         } else if (this.jsonRpcToDeploymentClassPathJava.contains(jsonRpcMethodName)) { // Route to extension (deployment)
             Object item = DevConsoleManager.invoke(jsonRpcMethodName, getArgsAsMap(jsonRpcRequest));
-            codec.writeResponse(s, jsonRpcRequest.getId(), item,
-                    MessageType.Response);
+
+            // Support for Mutiny is diffcult because we are between the runtime and deployment classpath.
+            // Supporting something like CompletableFuture that is in the JDK works fine
+            if (item instanceof CompletableFuture) {
```

Comment:
Use CompletionStage instead

Reply:
Cool. Let me change that. I might do the same (later) for publisher.